packages/docgen/README.md

```diff
@@ -0,0 +1,244 @@
+# `docgen`
+
+`docgen` helps you to generate the _public API_ of your code. Given an entry point file, it outputs the ES6 export statements and their corresponding JSDoc comments in human-readable format.
+
+Some characteristics:
+
+* If the export statement doesn't contain any JSDoc, it'll look up for JSDoc up to the declaration.
+* It can resolve relative dependencies, either files or directories. For example, `import default from './dependency'` will find `dependency.js` or `dependency/index.js`
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/docgen --save-dev
+```
+
+## Usage
+
+`node src/cli.js <entry-point.js>`
+
+This command will generate a file named `entry-point-api.md` containing all the exports and its JSDoc comments.
+
+### CLI options
+
+* **--formatter** `(String)`: A path to a custom formatter to control the contents of the output file. It should be a CommonJS module that exports a function that takes as input:
+  * *rootDir* `(String)`: current working directory as seen by docgen.
+  * *docPath* `(String)`: path of the output document to generate.
+  * *symbols* `(Array)`: the symbols found.
+* **--ignore** `(RegExp)`: A regular expression used to ignore symbols whose name match it.
+* **--output** `(String)`: Output file that will contain the API documentation.
+* **--debug**: Run in debug mode, which outputs some intermediate files useful for debugging
+* **--append** `(String)`: Title of the Markdown section to append the generated documentation to. It takes precedence over and bypasses the *--formatter* option.
+
+## Examples
+
+### Default export
+
+Entry point:
+
+```js
```

Comment:
Out of curiosity, do you use the tool to generate the output of those examples?
If not, maybe it would make sense to find a way to do it, format can change so it would be easier to keep it in sync :)

Reply:
Yeah, I manually created some examples and then generated this output. I agree a more streamlined process would be great.

Perhaps a potential way to improve this would be what you proposed at https://github.com/WordPress/gutenberg/pull/13856#discussion_r256843355 If we had a way to mark a Markdown file with tokens (instead of the current system that uses a section title) it'd be useful for this as well.

Reply:
I've implemented a new CLI option `--to-token` that allows inserting the autogenerated docs within the Markdown file.